services/src/pro/layers/postgres_layer_db.rs

```diff
@@ -159,6 +145,25 @@ where
 
         trans.execute(&stmt, &[&collection_id, &layer_id]).await?;
 
+        let stmt = trans
+            .prepare(
+                "
+        INSERT INTO permissions (role_id, permission, layer_id)
+        VALUES ($1, $2, $3) ON CONFLICT DO NOTHING;",
```

Comment:
Will this still raise an error?

Reply:
what if the permission is "Owner" but you want to set it to "Read" or maybe "Deny" (if we add that) this might silently fail and keep the old permission right?  So there should be a response. We could reject the change and return an error. Or we apply the change and return the old permission? (I have seen this pattern at many sql statements)

Reply:
> what if the permission is "Owner" but you want to set it to "Read" or maybe "Deny" (if we add that) this might silently fail and keep the old permission right? So there should be a response. We could reject the change and return an error. Or we apply the change and return the old permission?

we have this constraint: "CREATE UNIQUE INDEX ON permissions (role_id, permission, dataset_id);" which means the same role can't have the same  permission on the same resource multiple times. We could tighten that further and require for each role at most one row, so you could no longer have Read AND Owner, but only one. Shall I do that?

Then again, the access permission evaluation works across all roles of a user and it will happen lots of times that a user has read permission by one role and Owner by another. So I don't think it's too much of a problem.

Reply:
no thats fine. I just think that there should be a feedback when something has no effect.  Maybe add a TODO and then we can think about that?

Reply:
00b3a35085a1cb2935b26841b8c19cd4246b1ae4